### DISCUSSION TITLE
Laser

### DISCUSSION THREAD

**Discussion by pionex** [grblHAL/iMXRT1062#54] [Q&A]
I'm looking at the Teensy41 breakout boards and I am not seeing an additional PWM signal to add a laser.  I may be missing something, but the documentation for the boards look like everything is setup to run a laser or a spindle at any given time.  I am currently using FluidNC and I can define both tools and easily switch to the laser using an M6 command.  Is there no functionality like this yet in grbl_hal?

**Comment by terjeio** [comment]
>  Is there no functionality like this yet in grbl_hal?

[There is](https://github.com/grblHAL/Plugins_spindle), for now for switching between a ModBus controlled VFD and PWM. In addition to this there is work in progress for implementing support for multiple spindles via the `$` command word, LinuxCNC style.

Note that no drivers currently has support two (or more) PWM outputs, either driver support has to be added or MCU specific plugin(s) has to be written that registers the spindle with the core.
A plugin can be written to trap the M6 command to switch spindles when a specific tool number is asked for (I guess that is what FluidCNC does) without compromising the grblHAL [tool change](https://github.com/grblHAL/core/wiki/Manual,-semi-automatic-and-automatic-tool-change) functionality.
